modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/EnrichmentManager.scala

```diff
@@ -92,8 +88,53 @@ object EnrichmentManager {
                enriched.pii = pii.asString
              }
            }
+      _ <- validateEnriched(enriched, raw, processor, client)
     } yield enriched
 
+  private val atomicSchema: SchemaKey = SchemaKey("com.snowplowanalytics.snowplow", "atomic", "jsonschema", SchemaVer.Full(1, 0, 0))
+
+  private def validateEnriched[F[_]](
+    enriched: EnrichedEvent,
+    raw: RawEvent,
+    processor: Processor,
+    client: Client[F, Json]
+  )(
+    implicit
+    M: Monad[F],
+    L: RegistryLookup[F],
+    C: Clock[F]
+  ): EitherT[F, BadRow, Unit] =
+      enriched.toShreddedEvent
+        .leftMap(err =>
+          EnrichmentManager.buildEnrichmentFailuresBadRow(
+            NonEmptyList.one(
+              FailureDetails.EnrichmentFailure(Some(EnrichmentInformation(atomicSchema, identifier = "shredded_validator")),
```

Comment:
Would be nice to add a message in the bad row that says that the issue happened while validating the enriched event before sending.